asv_bench/benchmarks/io/csv.py

```diff
@@ -53,15 +53,32 @@ class TimeReadCsvSkiprows(BaseReadCsv):
         UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
         [
             None,
-            lambda x: x % 2,
-            np.arange(1, UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE][0][0] // 10),
-            np.arange(1, UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE][0][0], 2),
+            "lambda_even_rows",
+            "range_uniform",
+            "range_step2",
         ],
     ]
 
+    def setup(self, test_filenames, shape, skiprows):
+        super().setup(test_filenames, shape, skiprows)
+
+        if skiprows:
+            skiprows_dict = {
+                "lambda_even_rows": lambda x: x % 2,
+                "range_uniform": np.arange(
+                    1, UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE][0][0] // 10
+                ),
+                "range_step2": np.arange(
+                    1, UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE][0][0], 2
+                ),
+            }
```

Comment:
shouldn't it be a class attribute? if the number of parameters becomes bigger, the setup function became to be hard to read